### PR TITLE
DB-2378: fixing moz-ext URL in url bar

### PR DESCRIPTION
**Pull request by tsl143**
With this check
image

Without this check
image

### CODE REVIEW

File: mozilla-release/browser/components/urlbar/UrlbarInput.jsm [hunk 1]
```diff
@@ -1090,7 +1090,19 @@ class UrlbarInput {
 
     this.valueIsTyped = false;
     this._resultForCurrentValue = null;
-    this.inputField.value = val;
+
+    // DB-2378 => CLIQZ-SPECIAL: do not set url if freshtabURL exists,
+    // do not check if custom URL is set by user
+    const homeUrl = Services.prefs.getCharPref("browser.startup.homepage", "");
```
Comment (remusao) [inline mozilla-release/browser/components/urlbar/UrlbarInput.jsm]: What about reading the pref only if `val.startsWith("moz-ext")` (which I assume will be false in more than 99% of cases). Then there would be no overhead at all?

Comment (alver-cliqz) [reply mozilla-release/browser/components/urlbar/UrlbarInput.jsm]: then we will need to add one more condition

Comment (remusao) [reply mozilla-release/browser/components/urlbar/UrlbarInput.jsm]: What about:
```js
    // DB-2378 => CLIQZ-SPECIAL: do not set url if freshtabURL exists,
    // do not check if custom URL is set by user
    if (
      val.startsWith("moz-ext") &&
      val === Services.prefs.getCharPref("browser.startup.homepage", "")
    ) {
      this.inputField.value = "";
    } else {
      this.inputField.value = val;
    }
```

Comment (alver-cliqz) [reply mozilla-release/browser/components/urlbar/UrlbarInput.jsm]: I suggested:

Comment (alver-cliqz) [reply mozilla-release/browser/components/urlbar/UrlbarInput.jsm]: this.inputField.value = val;
    if (val.startsWith("moz-ext")) {
      const homeUrl = Services.prefs.getCharPref("browser.startup.homepage", "");
      if (val === homeUrl) {
        this.inputField.value = "";
      }
    }

Comment (remusao) [reply mozilla-release/browser/components/urlbar/UrlbarInput.jsm]: Up to you guys!

Comment (alver-cliqz) [reply mozilla-release/browser/components/urlbar/UrlbarInput.jsm]: Good idea anyway, we will fix this

**Pull request merged**